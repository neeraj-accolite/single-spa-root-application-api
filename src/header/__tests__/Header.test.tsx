import {describe, test} from 'vitest';
import {render,fireEvent,waitFor} from '@testing-library/react';
import Header from '../Header';

describe("Header React Micro-frontend application", ()=>{
    test("The header component is rendering correctly", ()=>{
        const component = render(<Header />);
        expect(component).toMatchSnapshot()
    });

    test("The profile data should be displayed if the event is received",async ()=>{
        const component = render(<Header />);
        // const elem1 =component.getByText("Profiles");
        // 
        fireEvent(window,new CustomEvent('profile-order-fetched',{detail :{
            user:{ 
                first_name: 'Test',
                last_name: 'User',
            },
            qty: 10
        } }));
        const elem = component.getByText("Test User");
        console.warn('aaa ',elem)
    });
})