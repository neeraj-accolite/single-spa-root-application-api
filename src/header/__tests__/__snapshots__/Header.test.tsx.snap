// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header React Micro-frontend application > The header component is rendering correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="header"
      >
        <img
          class="logoImage"
          src="https://www.nutrien.com/themes/custom/nutrien/logo.svg?ver=1721186473"
        />
        <div
          class="actionBar"
        >
          <nav>
            <ul
              id="menu"
            >
              <li>
                <a
                  href="/profiles"
                  id="menu"
                >
                  Profiles
                </a>
              </li>
              <li>
                <a
                  href="/address"
                  id="menu"
                >
                  Address
                </a>
              </li>
              <li>
                <a
                  href="/orders"
                  id="menu"
                >
                  Orders
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="header"
    >
      <img
        class="logoImage"
        src="https://www.nutrien.com/themes/custom/nutrien/logo.svg?ver=1721186473"
      />
      <div
        class="actionBar"
      >
        <nav>
          <ul
            id="menu"
          >
            <li>
              <a
                href="/profiles"
                id="menu"
              >
                Profiles
              </a>
            </li>
            <li>
              <a
                href="/address"
                id="menu"
              >
                Address
              </a>
            </li>
            <li>
              <a
                href="/orders"
                id="menu"
              >
                Orders
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
